flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -689,7 +693,7 @@ void scheduleOrUpdateConsumers(ResultPartitionID partitionId) {
 
 		if (partition.getIntermediateResult().getResultType().isPipelined()) {
 			// Schedule or update receivers of this partition
-			partition.markSomePipelinedDataProduced();
+			partition.markDataProduced();
```

Comment:
Shouldn't we always call `partition.markDataProduced` also if it's a `BLOCKING` result partition?

Reply:
Sure,  `partition.markDataProduced` is fine for BLOCKING result. I moved it out from the `if` clause.

In my understanding, the `ExecutionVertex.scheduleOrUpdateConsumers` is for PIPELINED partition currently. Only PIPELINED partition will send the `scheduleOrUpdateConsumers` message and the method will throw exception if the partition is BLOCKING.